SynoCommunity/spksrc
# FFMPEG 4.2.2: Update, fixes and cleanup

_Motivations:_  Previous patchset contained many unneded features & limitations now fixed by breaking into individual patches every synology changesets.  As well, porting some of the other options used by Synology in their default build.

_Linked issues:_   #3716, #3861, #3862, #3868 
  * ffmpeg conversion from FLAC to AAC not working #3861 - Fixed & Closed
  * Cannot convert .mp4 to .jpg #3862 - Fixed & Closed
  * dts wav failed due to `-write_info_tag` option #3716 - Presumed fixed & Closed
  * TVHeadend/Surveillance Station/DSCam #3868 - Fixed & Closed
  * TVheadend 4.3 for cedarview #3863 - **NOT Related**, Fixed in PR #3876
  * Comskip - codec problems #3878 - Fixed & Closed

### Changes
* Major clean-up of Synology patches against previous versions to cherry-pick interesting bits
* Update to latest stable
* Activation of neon acceleration in libswscale (should "boosts" transcoding performance on Alpine platform)
* Groundwork for `_smd` acceleration enablement in the future for `evansport` arch
* evansport specific platform compile time cpu arch setting
* Enablement of opencore-amr codec
* Update libva-utils to v2.6 + fix https://github.com/intel/libva-utils/pull/185
* `-O3` compile-time optimization
* Update libvpx from v1.6.1 to v1.8.2 and enable high bit depth

_Related Work:_
* `apparmor` limitations: #3847
* Automated VideoStation fixes: https://github.com/SynoCommunity/spksrc/pull/3828

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### IMPORTANT NOTES
* Due to the amount of patches this PR requires #3896 as dependancy
* Publishing of this updated package will require publishing new tvheadend packages as well in order to fix #3868 and other potential limitations due to previous version of the synology patchset applied.
* Contains groundwork in preparation for `evansport` native hardware acceleration.  The enablement of this feature is not yet in a readiness state to be applied.  Thus this pull request contains notes for future work related to `evansport` native hardware acceleration using built-in PowerVR GPU.
